components/src/lib/CancelButton/CancelButton.tsx

```diff
@@ -0,0 +1,21 @@
+import styles from './CancelButton.module.css';
+import { CancelButtonProps } from '../interfaces';
+
+const CancelButton = (props: CancelButtonProps) => {
+  return (
+    <button
+      className={styles['button-cancel']}
+      style={{
+        color: `${props.fontColor || 'black'}`,
+        backgroundColor: `${props.backgroundColor || 'white'}`,
+        width: `${(props.buttonSize && props.buttonSize.width) || '100%'}`,
```

Comment:
```suggestion
        width: `${(props.buttonSize?.width) ?? '100%'}`,
```

`&&` 대신 옵셔널 체이닝으로 표현할 수 있을까요? 사용자가 width로 number `0`를 지정할 경우, '100%'로 평가될 것 같아 같이 `??` 사용도 제안드려보아요

Reply:
말씀해주신 것처럼 `0`을 지정하면 `100%`로 평가됩니다!
처음부터 이 부분을 의도하고 코드를 작성하지는 않았지만, 사용자가 `0`을 입력하는 것을 일종의 버그라고 본다면 `100%`로 평가되게끔 하는 것이 적절할 것 같아요. width가 `0`인 경우라면 버튼을 보이지 않게 하고 싶은 경우인 것 같은데, 그럴 때는 buttonSize를 설정하기보다 아예 cancelButton prop을 넘기지 않는 것이 옳다고 생각해요.

### 질문
0과 같이 적절하지 않은 값일 때 `100%`로 평가하고 싶다면, 지금처럼 `&&`를 사용하는 것보다 별도로 0인지 확인하는 로직을 작성하는 것이 나을까요?

Reply:
> 사용자가 0을 입력하는 것을 일종의 버그라고 본다면 100%로 평가되게끔 하는 것이 적절할 것 같아요. 

아하 
음 그런데, 0에서 100% 로 애니메이션을 변화시키고 싶을 수도 있고..(있나..?) 굳이 막을 필요는 없지 않나 싶습니다. 
(0이 잘못된 값이라 막는다면 -1 도 막고 0.0314도 막아야할 것 같은데 우연히 0만 막히는 것은 조금 어색해보입니다.)

0을 넣었는데 100%으로 처리되는 동작을 유지해주신다면, 
(사용자 입장에서 예측 가능한 동작은 아닌 것 같아) 주석으로 또는 개발 문서로 해당 동작을 사용자에게도 알려주어야 한다고 생각합니다. 

> 0과 같이 적절하지 않은 값일 때 100%로 평가하고 싶다면, 지금처럼 &&를 사용하는 것보다 별도로 0인지 확인하는 로직을 작성하는 것이 나을까요?

(`&&`는 어느 부분인지 제가 잘 파악하지 못했는데 `||`를 의미해주신게 맞을까요?)

코드 상에서 0이 올 수 있는 값 (여기서는 width) 오른쪽에 `||` 연산자를 쓴 코드는, 작지 않은 확률로 falsy값일 때의 케이스를 놓친 코드로 보입니다. 따라서 `||` 보다는 `??`를 제안드립니다.

말씀해주신 것처럼 "적절하지 않은 값"일 때 모두를 대응하여 '100%'으로 처리하실 계획이시라면 `isInvalidWidth(width) ? '100%' : width`처럼 코드 상에 드러나면 좋을 것 같네요 :)